### PR TITLE
chore(Root): upgrade node version

### PR DESCRIPTION
**Pull request by kikoruiz**
Upgrade Node version (to __v20__).
Also **start versioning `package-lock.json`** file and add **cache to GHA** main workflow installations.

### CODE REVIEW

**Review by nucliweb — APPROVED** [review]

**Review by jordevo — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -64,12 +66,12 @@ jobs:
         with:
           registry-url: 'https://registry.npmjs.org'
           node-version-file: '.nvmrc'
+          cache: 'npm'
 
       - name: Install
-        run: npm install --no-save --no-fund --no-audit --legacy-peer-deps
+        run: npm ci --no-save --no-fund --no-audit --legacy-peer-deps
```
Comment (jordevo) [inline .github/workflows/main.yml:72]:
```suggestion
        run: npm ci
```

these are `npm install` flags, not `npm ci`'s, right? ` --no-save --no-fund --no-audit --legacy-peer-deps`

Comment (kikoruiz) [reply .github/workflows/main.yml:72]: They are also valid for `ci`: https://docs.npmjs.com/cli/v10/commands/npm-ci#configuration

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -12,12 +12,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Cancel Previous Redundant Builds
-        uses: styfle/cancel-workflow-action@0.6.0
+        uses: styfle/cancel-workflow-action@0.12.1
```
Comment (andresz1) [inline .github/workflows/main.yml:15]: Cancel workflow is not needed anymore 🙏🏻  you can do this instead
```
concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
```

File: .github/workflows/main.yml (hunk 1 above)
Comment (andresz1) [reply .github/workflows/main.yml:72]: Yes, we need them. We can also add them in the `npmrc`

**Review by andresz1 — APPROVED** [review]

**Review by carlosvillu — APPROVED** [review]

File: .github/workflows/main.yml (hunk 2 above)
Comment (kikoruiz) [reply .github/workflows/main.yml:15]: Right!

**Pull request merged**